### PR TITLE
Bumps frontend-build@^14, frontend-platform@^8

### PR DESCRIPTION
**Pull request by pomegranited**
Upgrades dependencies to support parallel upgrades to frontend-app-course-authoring.

### Supporting information

https://github.com/openedx/frontend-app-course-authoring/issues/1034

Private-ref: [FAL-3750](https://tasks.opencraft.com/browse/FAL-3750)

### Author Notes & Concerns

1. This is my first attempt at updating npm dependencies, so any advice is welcome :)
1. This package is a dependency of many other frontend packages, so I'm unsure how to test the effects of this change.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @pomegranited! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by pomegranited** [comment]
Hi @mphilbrick211 , I don't know if there are any CCs with merge rights on this repo? But it's blocking us merging https://github.com/openedx/frontend-app-course-authoring/pull/1052, so it's getting urgent. Could someone from Axim help out?

**Comment by mphilbrick211** [comment]
Hi @pomegranited! I'll check for you.

**Comment by arbrandes** [comment]
@pomegranited, I don't think anybody outside 2U is set up to properly review or test this.  I'm actually surprised this repository is even a thing, let alone a dependency in a core repo.  I'm going to argue for its deprecation/removal.

In the meantime, it seems @hajorg is a major contributor here.  Could you take a look, please?

File: package.json
Comment (BrandonHBodine) [inline package.json:10]: What is the purpose of this `prepare` script? It's identical to the `build` one.

**Review by BrandonHBodine — APPROVED** [review]
lgtm. Left a question.

**Comment by BrandonHBodine** [comment]
Hey @arbrandes, I think this was our answer to getting something in before the introduction of the plugin framework. It currently is a pain on the 2u side as well. Do you know where the discussions for a adding a plugin slot to the course authoring MFE take place?

**Comment by arbrandes** [comment]
@BrandonHBodine, thanks for reviewing!

> Do you know where the discussions for a adding a plugin slot to the course authoring MFE take place?

There are a few places to start an conversation about that.  The #wg-frontend channel in the Open edX Slack is a good one, but the [FWG meeting](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3593961516/Frontend+Working+Group+Meeting+Agendas+and+Minutes) every other Thursday also works.  So would the forum, for that matter.

But the _fastest_ way is usually to open a PR - in this case to frontend-app-course-authoring - where we can iterate on a solution quickly.

File: package.json
Comment (pomegranited) [reply package.json:10]: Hi @BrandonHBodine -- this change is optional, but it lets us reference unmerged branches when working on dependency upgrades [like this](https://github.com/openedx/frontend-app-course-authoring/pull/1052/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R48).

> Since npm v5.0.0 ([ref](https://blog.npmjs.org/post/161081169345/v500)):
> 
>> git dependencies with `prepare` scripts will have their `devDependencies` installed, and `npm install` run in their directory before being packed.

**Comment by pomegranited** [comment]
Thank you for your review @BrandonHBodine ! Could this be merged and tagged so we can unblock https://github.com/openedx/frontend-app-course-authoring/pull/1052? CC @arbrandes

**Comment by openedx-webhooks** [comment]
@pomegranited 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-component-ai-translations&entry.867055334=https://github.com/openedx/frontend-component-ai-translations/pull/16&entry.1484655318=https://github.com/pomegranited&entry.752974735=2024-05-28+02:11&entry.1917517419=2024-06-20+13:42&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by bradenmacdonald** [comment]
Thanks @BrandonHBodine!

**Pull request merged**
